src/benchmarklib/benchmark_utils.cpp

```diff
@@ -330,6 +331,43 @@ nlohmann::json EncodingConfig::to_json() const {
   return json;
 }
 
+void BenchmarkTableEncoder::encode(const std::string& table_name, std::shared_ptr<Table> table,
+                                   const EncodingConfig& config) {
+  const auto& type_mapping = config.type_encoding_mapping;
+  const auto& custom_mapping = config.custom_encoding_mapping;
+
+  const auto& column_mapping_it = custom_mapping.find(table_name);
+  const auto table_has_custom_encoding = column_mapping_it != custom_mapping.end();
+
+  ChunkEncodingSpec chunk_spec;
+
+  for (ColumnID column_id{0}; column_id < table->column_count(); ++column_id) {
+    if (table_has_custom_encoding) {
+      const auto& column_name = table->column_name(column_id);
+      const auto& column_mapping = column_mapping_it->second;
+      const auto& column_encoding = column_mapping.find(column_name);
+      if (column_encoding != column_mapping.end()) {
+        // The column has a custom encoding
+        chunk_spec.push_back(column_encoding->second);
+        continue;
+      }
+    }
+
+    const auto& column_type_str = data_type_to_string.left.find(table->column_data_type(column_id))->second;
+    const auto& type_encoding = type_mapping.find(column_type_str);
```

Comment:
Why do the mapping via std::strings when we have the (safer) enum `DataType`? 
Again, I'd prefer "encoding_by_data_type" over "type_mapping"

Reply:
The string thing is probably already in the master, right? mmpf

Reply:
In the master because its primary use case is as a json config for benchmarks. And all we get there are strings...

Reply:
That sounds fairly reasonable. I'll do it for the `DataType`s in a follow up PR.